src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -143,17 +146,147 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
   }
+
+  // identifier constants
+  private final int MARKDOWN_BOLDITALIC_IDENTIFIER_LENGTH = "***".length();
+  private final int MARKDOWN_BOLD_IDENTIFIER_LENGTH = "**".length();
+  private final int MARKDOWN_ITALIC_IDENTIFIER_LENGTH = "*".length();
+  private final int BBCODE_COLOR_BEGIN_IDENTIFIER_LENGTH = "\\[color=".length() - 1;
+  private final int BBCODE_COLOR_END_IDENTIFIER_LENGTH = "\\[/color\\]".length() - 2;
+  private final int BBCODE_SIZE_BEGIN_IDENTIFIER_LENGTH = "\\[size=".length() - 1;
+  private final int BBCODE_SIZE_END_IDENTIFIER_LENGTH = "\\[/size\\]".length() - 2;
+
+  /**
+   * this function replaces a word surrounded by ** with a bold HTML tag
+   */
+  public String markDownBoldToHTML(String markdownChunk) {
+    String content = markdownChunk.substring(MARKDOWN_BOLD_IDENTIFIER_LENGTH,
+        markdownChunk.length() - MARKDOWN_BOLD_IDENTIFIER_LENGTH);
+    return String.format("<b>%s</b>", content);
+  }
+
+  /**
+   * this function replaces a word surrounded by * with an italics HTML tag
+   */
+  public String markDownItalicToHTML(String markdownChunk) {
+    String content = markdownChunk.substring(MARKDOWN_ITALIC_IDENTIFIER_LENGTH,
+        markdownChunk.length() - MARKDOWN_ITALIC_IDENTIFIER_LENGTH);
+    return String.format("<i>%s</i>", content);
+  }
+
+  /**
+   * this function replaces a word surrounded by *** with a bold-italics HTML
+   * tag
+   */
+  public String markDownBoldItalicToHTML(String markdownChunk) {
+    String content = markdownChunk.substring(MARKDOWN_BOLDITALIC_IDENTIFIER_LENGTH,
+        markdownChunk.length() - MARKDOWN_BOLDITALIC_IDENTIFIER_LENGTH);
+    return String.format("<b><i>%s</i></b>", content);
+  }
+
+  /**
+   * this function replaces words surrounded with [color="<COLOR>"] ... [/color]
+   * with a color HTML tag
+   */
+  public String bbCodeColorToHTML(String markdownChunk) {
+    String content = markdownChunk.substring(BBCODE_COLOR_BEGIN_IDENTIFIER_LENGTH,
+        markdownChunk.length() - BBCODE_COLOR_END_IDENTIFIER_LENGTH);
+    String[] colorContent = content.split("\\]", 2);
+    String color = colorContent[0];
+    content = colorContent[1];
+    return String.format("<font color=\"%s\">%s</font>", color, content);
+  }
+
+  /**
+   * this function replaces words surrounded with [size="<SIZE>"] ... [/size]
+   * with a font size HTML tag
+   */
+  public String bbCodeSizeToHTML(String markdownChunk) {
+    String content = markdownChunk.substring(BBCODE_SIZE_BEGIN_IDENTIFIER_LENGTH,
+        markdownChunk.length() - BBCODE_SIZE_END_IDENTIFIER_LENGTH);
+    String[] sizeContent = content.split("\\]", 2);
+    String size = sizeContent[0];
+    content = sizeContent[1];
+    return String.format("<font size=\"%s\">%s</font>", size, content);
+  }
+
+  /**
+   * this function replaces *, **, ***, [color="<COLOR>] ... [/color], and [size="<SIZE>"] ... [/size]
+   * identifiers with the appropriate HTML
+   * tag
+   */
+  public String markDownToHTML(String message) {
```

Comment:
this is not a big deal at all: In line 150 there already a variable of type Message with name message. Even though this message variable here will ofc work with the same name, maybe change the name slightly so that someone reading through is clear that it's a different thing?